flexmeasures/data/services/accounts.py

```diff
@@ -0,0 +1,19 @@
+from typing import List, Optional
+
+from flexmeasures.data.models.user import Account, AccountRole
+
+
+def get_accounts(
+    role_name: Optional[str] = None,
+) -> List[Account]:
+    """Return a list of Account objects.
+    The role_name parameter allows to filter by role.
+    """
+    account_query = Account.query
+
+    if role_name is not None:
+        role = AccountRole.query.filter(AccountRole.name == role_name).one_or_none()
+        if role:
+            account_query = account_query.filter(Account.account_roles.contains(role))
```

Comment:
Looks as if this would return all accounts in case the role name is not in the database, which is not what I would consider to be the expected behavior.

It should also be possible to rewrite this as a single query, but the speedup is probably not significant in the larger scheme of things.

Reply:
Good catch. This feature is not used now, but important to get it right.